build_unix/Framework/CMakeLists.txt

```diff
@@ -0,0 +1,201 @@
+cmake_minimum_required(VERSION 2.8)
+project(Framework)
+
+set(CMAKE_MODULE_PATH
+	${CMAKE_CURRENT_SOURCE_DIR}/../../../Dependencies/cmake-modules
+	${CMAKE_MODULE_PATH}
+)
+
+if(NOT CMAKE_BUILD_TYPE)
+	set(CMAKE_BUILD_TYPE Release CACHE STRING
+		"Choose the type of build, options are: None Debug Release"
+		FORCE)
+endif()
+
+if(CMAKE_BUILD_TYPE STREQUAL "Release")
+	add_definitions(-DNDEBUG)
+else()
+	add_definitions(-D_DEBUG)
+endif()
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
+if(APPLE)
+	set(CXX_FLAGS "${CMAKE_CXX_FLAGS} -x objective-c++")
+endif(APPLE)
+
+find_package(ZLIB)
+if (NOT ZLIB_FOUND)
+	MESSAGE("-- Using Provided zlib source")
+	if(NOT TARGET zlibstatic)
+		add_subdirectory(
+			${CMAKE_CURRENT_SOURCE_DIR}/../../../Dependencies/zlib-1.2.8
+			${CMAKE_CURRENT_SOURCE_DIR}/../../../Dependencies/zlib-1.2.8/build_${CMAKE_SYSTEM_NAME}
+		)
+	endif()
+	include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../../../Dependencies/zlib-1.2.8)
+	include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../../../Dependencies/zlib-1.2.8/build_${CMAKE_SYSTEM_NAME})
+	list(APPEND PROJECT_LIBS zlibstatic)
+else()
+	include_directories(${ZLIB_INCLUDE_DIRS})
+	list(APPEND PROJECT_LIBS ${ZLIB_LIBRARIES})
+endif()
+
+if (NOT TARGET Boost::boost)
+	add_subdirectory(
+		${CMAKE_CURRENT_SOURCE_DIR}/../../../Dependencies/boost-cmake
+		${CMAKE_CURRENT_SOURCE_DIR}/../../../Dependencies/boost-cmake/build_${CMAKE_SYSTEM_NAME}
+	)
+endif()
+
+if(ANDROID)
+	if(NOT TARGET cpufeatures)
+		include_directories("${CMAKE_ANDROID_NDK}/sources/android/cpufeatures")
+	endif()
+endif()
+
+if (WIN32)
+	add_definitions(-DWIN32)
+	add_definitions(-D_UNICODE)
```

Comment:
the windows build was erroring out without _UNICODE, so not sure if UNICODE is required, but I've added it in, since it doesn't seem to effect the build anyway.